### DISCUSSION TITLE
Compiling lynx with cowasm

### DISCUSSION THREAD

**Discussion by sandkoan** [sagemathinc/cowasm#52] [Q&A]
This seems like a really compelling idea. Recently, I've been trying to compile `lynx` to wasm, but have been running into issues with the ncurses library (which, it doesn't seem, emscripten has support for). After some searching, I found cowasm, which seems to have ncurses packages.

How would I go about compiling `lynx` to wasm using cowasm's ncurses?

**Comment by williamstein** [comment]
I haven't looked into lynx yet.  However, probably the easiest way now would be to start with the rogue package here

https://github.com/sagemathinc/cowasm/tree/main/packages/rogue

which also uses ncurses.   First ensure you can build it on your computer by typing "make".

Then copy packages/rogue to a new directory packages/lynx, make changes, and go from there.

I assume you're talking about the lynx web browser, in which case presumably networking will be the really interesting challenge.  Where are you trying to run lynx?

**Reply by sandkoan** [reply]
I am indeed referring to the lynx web browser.

I'm trying to build a client-side version of [text-mode](https://earthly-tools.com/text-mode): https://earthly.dev/blog/text-mode/, and for that I need to be able to run the `lynx` binary in the browser—hence the need to compile it to wasm.

I only plan to run `lynx` with the `--dump` flag, which doesn't even require the curses library, but I don't know how to extract the necessary non-curses parts, so I'm forced to compile the entire thing to wasm.

I'm following the steps outlined [here](https://gist.github.com/Potherca/866dfc72de9bfe0fc5627945446826a6).

**Reply by williamstein** [reply]
Will you need a TCP network stack for this?  Or are you just browsing files via lynx?

**Reply by sandkoan** [reply]
I won't need a TCP network stack, because I expect to already have access to the underlying HTML/DOM, so it just becomes a matter of dumping/converting said HTML to raw plaintext, which is, really, the only thing I need `lynx` for.
